python/ray/tune/trial.py

```diff
@@ -388,6 +391,20 @@ def update_last_result(self, result, terminate=False):
         self.last_result = result
         self.last_update_time = time.time()
         self.result_logger.on_result(self.last_result)
+        for metric, value in result.items():
```

Comment:
@hershg weren't we going to do `min-last`, `max-last`, `min-all`, `max-all`?

Reply:
oh right, just pushed a change for this. should we still have "max" and "min" as an option since the previous code using analysis expect to have {metric, mode=[max, min]} so this change would break that?

Also would having mode of "max-avg" and "min-avg" be a good feature? (within each trial, store the avg; based on max- or min- prefix choose "best" trial)

Reply:
I think it should be {metric, mode=[max, min], scope=[all, last]}

I don't think we need `avg` - though shouldn't be hard to add